### ISSUE TITLE
Cannot link dashboard filters to Custom questions with filtered aggregate data

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#11007]
**Describe the bug**
When creating a Custom question and doing filter aggregation on returned data (generating a sub-select), then dashboard filters cannot be connected to card.

**Workaround**
Include another question with the same columns, which isn't aggregated, and also connect the filters to that question. That will make the dashboard filters show correctly.

**To Reproduce**
1. Create Custom question from Sample Data > Orders
2. Filter `User ID = 1`, Summarize `Sum of Total` grouped by `Created At: Day` and `Product ID` and `Product -> Category`
3. Aggregation filter `sum is greater than 100`
![image](https://user-images.githubusercontent.com/1447303/83942260-564c8100-a7f2-11ea-891a-4cf5629950b9.png)
4. Save question and add to dashboard
5. Add filters `Time: All options` and `ID` and `Other Categories` and try to connect to question.

<details>
  <summary>Generated SQL - click to expand</summary>
SELECT "source"."CREATED_AT" AS "CREATED_AT", "source"."PRODUCT_ID" AS "PRODUCT_ID", "source"."CATEGORY" AS "CATEGORY", "source"."sum" AS "sum"
FROM (SELECT CAST("PUBLIC"."ORDERS"."CREATED_AT" AS date) AS "CREATED_AT", "PUBLIC"."ORDERS"."PRODUCT_ID" AS "PRODUCT_ID", "PRODUCTS__via__PRODUCT_ID"."CATEGORY" AS "CATEGORY", sum("PUBLIC"."ORDERS"."TOTAL") AS "sum" FROM "PUBLIC"."ORDERS"
LEFT JOIN "PUBLIC"."PRODUCTS" "PRODUCTS__via__PRODUCT_ID" ON "PUBLIC"."ORDERS"."PRODUCT_ID" = "PRODUCTS__via__PRODUCT_ID"."ID"
WHERE "PUBLIC"."ORDERS"."USER_ID" = 1
GROUP BY CAST("PUBLIC"."ORDERS"."CREATED_AT" AS date), "PUBLIC"."ORDERS"."PRODUCT_ID", "PRODUCTS__via__PRODUCT_ID"."CATEGORY"
ORDER BY CAST("PUBLIC"."ORDERS"."CREATED_AT" AS date) ASC, "PUBLIC"."ORDERS"."PRODUCT_ID" ASC, "PRODUCTS__via__PRODUCT_ID"."CATEGORY" ASC) "source" WHERE "source"."sum" > 100
LIMIT 1048576
</details>

**Expected behavior**
Ability to do filtering via dashboard.

**Information about your Metabase Installation:**
Metabase 0.33.3 and release-0.33.x

**Additional context**
See forum thread for real database queries:
https://discourse.metabase.com/t/multiple-filters-in-dashboards/7487
Related to #9802 and #12654

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by salsakran** [comment]
My read is that there is no error thrown, but rather that the post aggregation filter isn't available in the dimension picker.

As such, it's closer to our new "completeness" tag. This is meant to signify that something that seems like it should be possible given the building blocks the interface offers.

**Comment by rthouvenin** [comment]
For anyone blocked by this issue and looking for a workaround, I described one in #11559

**Comment by AmirTKashi** [comment]
I have a similar issue when trying to link filters to Custom questions. I can formulate the issue as follows:


### Background
___
When creating a Custom question and doing another filter or aggregation on returned data (generating a sub-select or aggregate), then dashboard filters cannot be connected to the card.


### To reproduce
___
Create two custom questions as presented below:

**Custom Question 1:**

![Q1](https://user-images.githubusercontent.com/40166454/74733590-8886a280-5244-11ea-80c5-4f2cb3d849e3.png)


**Custom Question 2:**

![Q2](https://user-images.githubusercontent.com/40166454/74733374-19a94980-5244-11ea-9e4d-e087e38d699e.png)

(the only difference is that the second question has an extra filter at the end)


Save and add them to a dashboard.

![Dash1](https://user-images.githubusercontent.com/40166454/74733755-ea470c80-5244-11ea-8133-b2cf6113a6cc.png)




Then add an ID filter to the dashboard and try to link the filter to the cards. You will get this:

![dash-filter](https://user-images.githubusercontent.com/40166454/74733774-f501a180-5244-11ea-8886-96d8dc4bcb88.png)

(the second question can't be linked to the filter. The same issue occurs if you have a second aggregation instead of the second filter).


### Expected solution
___
I want to be able to link the filter to the first level of the data in the custom question GUI (I can do this query with the expected filter with the Native Query).


### Alternative
___
The SQL is straightforward, but I would prefer using the GUI.


### Severity
___
It's prevents non SQL-savvy user to create "Year on Year" charts. But I'd say it's rather a nice-to-have.


### Context
___
Metabase v0.34.0

**Comment by andreychirikba** [comment]
The same problem manifests itself when trying to apply a dashboard filter to an aggregated query that is based on a saved question:

1. Create a saved question.
2. Create a new question based on it and apply aggregation to it.
3. Add the question to a dashboard.
4. Create a dashboard filter.
5. Observe the field from the saved question is allowed to be selected, but no values are gathered from it into the filter dropdown and inputting any value into the filter manually results in an error.

This scenario works well, though, when, instead of starting from a saved question in (2), I modify the saved question (apply aggregation to it) and continue with its modified version further.

**Comment by flamber** [comment]
This is fixed on `master` - pretty sure it was #12531

**Comment by flamber** [comment]
Reopening. Something changed between when I tested and 0.36.0

**Comment by nemanjaglumac** [comment]
This one seems to be fixed.
Writing a repro for it to confirm.

**Comment by flamber** [comment]
This was fixed in 0.37.0, but it doesn't show a dropdown/search, which is covered in #12985. And now there's a test for this.

**Comment by flamber** [comment]
@okcomputerik
I don't see how you have been able to connect a dashboard filter to the question you are showing, since that would require that the first aggregation contained the email column, but that would give you different results.

My guess is that the email column has been part of the first aggregation in the question at some point, where it was connected to a filter on the dashboard, but then later the question was change, which results it a broken query.

It has never been possible to select which aggregation (section) a dashboard filter should filter on. I think you would likely need to create your question in SQL to do what you're trying to.

But you are not seeing this issue, since that is fixed and has several tests to prevent regressions.

Please use the forum for questions and troubleshooting: https://discourse.metabase.com/
